File: app/graphql/gql/queries/session.rb
# Copyright (C) 2012-2025 Zammad Foundation, https://zammad-foundation.org/

module Gql::Queries
  class Session < BaseQuery

    description 'The sessionId of the currently authenticated user.'

    type Gql::Types::SessionType, null: false

    def resolve(...)
      {
        id:         context[:sid],
        after_auth: Auth::AfterAuth.run(context.current_user, context[:controller].session)
      }
    end

  end
end

File: app/frontend/tests/graphql/factories/Session.ts
// Copyright (C) 2012-2024 Zammad Foundation, https://zammad-foundation.org/

import type { Session } from '#shared/graphql/types.ts'
import type { DeepPartial } from '#shared/types/utils.ts'

export default (): DeepPartial<Session> => {
  return {
    id: '6605e8986992bf38b8a03638a5c6090e',
    afterAuth: null,
  }
}
